frame/staking/src/tests.rs

```diff
@@ -1963,124 +1964,126 @@ fn phragmen_should_not_overflow_validators() {
 
 // TODO: overflow, should be checked
 // thread 'tests::phragmen_should_not_overflow_nominators' panicked at 'attempt to add with overflow'
-// #[test]
-// fn phragmen_should_not_overflow_nominators() {
-// 	ExtBuilder::default().nominate(false).build().execute_with(|| {
-// 		let _ = Staking::chill(Origin::signed(10));
-// 		let _ = Staking::chill(Origin::signed(20));
-//
-// 		bond_validator(2, StakingBalance::RingBalance(u64::max_value() / 2));
-// 		bond_validator(4, StakingBalance::RingBalance(u64::max_value() / 2));
-//
-// 		bond_nominator(6, StakingBalance::RingBalance(u64::max_value()), vec![3, 5]);
-// 		bond_nominator(8, StakingBalance::RingBalance(u64::max_value()), vec![3, 5]);
-//
-// 		start_era(1);
-//
-// 		assert_eq_uvec!(validator_controllers(), vec![4, 2]);
-//
-// 		// Saturate.
-// 		assert_eq!(Staking::stakers(3).total_power, u64::max_value());
-// 		assert_eq!(Staking::stakers(5).total_power, u64::max_value());
-// 	})
-// }
+#[test]
+fn phragmen_should_not_overflow_nominators() {
+	ExtBuilder::default().nominate(false).build().execute_with(|| {
+		let _ = Staking::chill(Origin::signed(10));
+		let _ = Staking::chill(Origin::signed(20));
+
+		bond_validator(2, StakingBalance::RingBalance(1));
+		bond_validator(4, StakingBalance::KtonBalance(1));
+
+		bond_nominator(6, StakingBalance::RingBalance(CAP), vec![3, 5]);
+		bond_nominator(8, StakingBalance::KtonBalance(CAP), vec![3, 5]);
+
+		start_era(1);
+
+		assert_eq_uvec!(validator_controllers(), vec![4, 2]);
+
+		// Saturate.
+		assert_eq!(Staking::stakers(3).total_power, TOTAL_POWER / 2);
+		assert_eq!(Staking::stakers(5).total_power, TOTAL_POWER / 2);
+	})
+}
```

Comment:
The `overflow  problem` TODOs mainly caused by using `u64::max_value()` but not `CAP` in darwinia-staking's `mock` file.

I think the `overflow tests` might point to 3 situations, but don't know if they are correct:

1. `bond_validator` reaches the max.
  a. `bond_validator`'s balance argument is the `CAP`
  b. `bond_nomnator`'s balance argument greater than zero.
2. `bond_nomnator` reaches the max.
  a. `bond_validator`'s balance argument is greater than zero.
  b. `bond_nomnator`'s balance argument is the `CAP`
3. both `bond_validator` and `bond_nomnator` reach the max.
  a. `bond_validator`'s balance argument is the `CAP`
  b. `bond_nomnator`'s balance argument is the `CAP`